sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -72,6 +72,68 @@ abstract class DataType extends AbstractDataType {
   /** The pretty (i.e. indented) JSON representation of this data type. */
   def prettyJson: String = pretty(render(jsonValue))
 
+  /**
+   * Util to recursively form JSON representation of data type, used for DESCRIBE AS JSON. Differs
```

Comment:
Can we move it to the command of DESCRIBE AS JSON? `DataType` is a public API and we should avoid polluting it.